src/Network/Wai/SAML2.hs

```diff
@@ -116,16 +118,22 @@ saml2Callback cfg callback app req sendResponse = do
 
             case lookup "SAMLResponse" body of
                 Just val -> do
-                    result <- validateResponse cfg val
                     let rs = lookup "RelayState" body
+                    result <- validateResponse cfg val <&>
+                                  fmap (\(assertion, response) ->
+                                          Result
+                                          { assertion = assertion,
+                                            relayState = rs,
+                                            response = response})
```

Comment:
Something like this to be in more in line with the style used:
```suggestion
                                          Result{ 
                                              assertion = assertion,
                                              relayState = rs,
                                              response = response
                                          })
```
or alternatively
```suggestion
                                          Result{ 
                                              assertion = assertion
                                          ,   relayState = rs
                                          ,   response = response
                                          })
```